osf/migrations/0043_add_quickfiles.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11 on 2017-06-13 17:35
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations, models
+from django.core.paginator import Paginator
+
+from osf.models import OSFUser, QuickFiles, Contributor
+from osf.models.quickfiles import get_quickfiles_project_title
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def add_quickfiles(*args, **kwargs):
+    ids_without_quickfiles = list(OSFUser.objects.exclude(created__type='osf.quickfiles').values_list('id', flat=True))
+
+    users_without_quickfiles = OSFUser.objects.filter(id__in=ids_without_quickfiles).order_by('id')
+    total_quickfiles_to_create = users_without_quickfiles.count()
+
+    logger.info('About to add QuickFiles for {} users.'.format(total_quickfiles_to_create))
+
+    paginated_users = Paginator(users_without_quickfiles, 1000)
+
+    total_created = 0
+    for page_num in paginated_users.page_range:
+        quickfiles_to_create = []
+        for user in paginated_users.page(page_num).object_list:
+            quickfiles_to_create.append(
+                QuickFiles(
+                    title=get_quickfiles_project_title(user),
+                    creator=user
+                )
+            )
+            total_created += 1
+
+        all_quickfiles = QuickFiles.objects.bulk_create(quickfiles_to_create)
+
+        contributors_to_create = []
+        for quickfiles in all_quickfiles:
+            quickfiles.add_addon('osfstorage', auth=None, log=False)
+            quickfiles.save()
+
+            contributors_to_create.append(
```

Comment:
It totally does when you use the shell! It's the post-save signal that adds the creator as contributor for QuickFiles, and that doesn’t get called in the migration. Found this little bug when working with Henrique who was using the current state of the QuickFiles backend. So this does add a step but as far as I can tell is actually needed